pkg/utils/utils.go

```diff
@@ -346,3 +372,47 @@ func Chroot(path string) (func() error, error) {
 		return syscall.Chroot(".")
 	}, nil
 }
+
+func RunCommand(command string, args ...string) (string, error) {
+	glog.V(2).Infof("RunCommand(): %s %v", command, args)
+	var stdout, stderr bytes.Buffer
+
+	cmd := exec.Command(command, args...)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	err := cmd.Run()
+	glog.V(2).Infof("RunCommand(): %s", stdout.String())
+	return stdout.String(), err
+}
+
+func GetNicSriovMode(pciAddress string) (string, error) {
+	glog.V(2).Infof("GetNicSriovMode(): device %s", pciAddress)
+	devLink, err := netlink.DevLinkGetDeviceByName("pci", pciAddress)
+	if err != nil {
+		return "", err
+	}
+	return devLink.Attrs.Eswitch.Mode, nil
+}
+
+func setNicSriovMode(pciAddress, mode string) error {
+	glog.V(2).Infof("setNicSriovMode(): device %s mode %s", pciAddress, mode)
+	devLink, err := netlink.DevLinkGetDeviceByName("pci", pciAddress)
+	if err != nil {
+		return err
+	}
+	if err := netlink.DevLinkSetEswitchMode(devLink, mode); err != nil {
+		return err
+	}
+	pfName, err := dputils.GetPfName(pciAddress)
+	if err != nil {
+		return err
+	}
+	offloadStat := "off"
+	if mode == "switchdev" {
+		offloadStat = "on"
+	}
+	args := []string{"-K", pfName, "hw-tc-offload", offloadStat}
+	_, err = RunCommand("ethtool", args...)
```

Comment:
The `ethtool` may not available in the image. It's better to use `chroot` to run this command from host instead.

Reply:
It'd be good to install it in the image by updating the docker file for the config daemon

Reply:
@pliurh ethool may not be also in the host